cpp/pipelines/add_authority_wikidata_ids.cc

```diff
@@ -0,0 +1,195 @@
+/** \file WikidataTest.cc
+ *  \brief functionality to acquire wikidata id corresponding to their gnds
+ *  \author andreas-ub
+ *
+ *  \copyright 2021 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "util.h"
+#include "StringUtil.h"
+#include "MARC.h"
+#include <iostream>
+#include <fstream>
+#include <regex>
+
+//Scenario 1:
+// use --create_mapping_file parameter <fileparth> to generate the mapping file 
+//  of the downloaded dnb authoriy dump (must bei unzipped first)
+//  Download from:  https://data.dnb.de/opendata/authorities-person_lds.jsonld.gz 
+//  and unzip to e.g. authorities-person_lds_20210613.jsonld
+//  output is stdout
+//Scenario 2:
+// use converted file from scenario 1 from cpp/data to create a map during 
+//  pipeline processing. The norm_data_input is extended by wikidata ids where possible
+//  and saved to 024 field indicator1:7 where wikidata id is not yet present
+[[noreturn]] void Usage() {
+    ::Usage("norm_data_marc_input norm_data_marc_output mapping_txt_file OR_USE_THIS_TO_CREATE_MAPPING_FILE: --create_mapping_file dnb_input_unzipped_file mapping_txt_file");
```

Comment:
Instead of OR_USE_THIS_TO_CREATE_MAPPING_FILE, please move the two invocation modes to different lines (maybe similar to installer.cc)

Also, if a program can be called in multiple modes, it might be more consistent to always require a first parameter that explicitly sets the mode (also like installer.cc) but i'm not absolutely sure about this one.

Reply:
In addition, it would be nice to move the comments into the usage output.